src/pcp/zoneinfo/pcp-zoneinfo.py

```diff
@@ -290,16 +298,54 @@ def __init__(self):
         self.pmSetLongOptionVersion()
         self.samples = None
         self.context = None
+def updateZoneinfoDataToBeFetched():
+    ZONEINFO_NUMBER_ZONE.update({
+        "nr_zone_active_anon"           :   "mem.zoneinfo.nr_zone_active_anon",
+        "nr_zone_inactive_file"         :    "mem.zoneinfo.nr_zone_inactive_file",
+        "nr_zone_active_file"           :   "mem.zoneinfo.nr_zone_active_file",
+        "nr_zone_unevictable"           :   "mem.zoneinfo.nr_zone_unevictable",
+        "nr_zone_write_pending"         :   "mem.zoneinfo.nr_zone_write_pending",
+        "nr_zspages"                    :   "mem.zoneinfo.nr_zspages",
+        "numa_local"                    :   "mem.zoneinfo.numa_local",
+        "nr_zone_inactive_anon"         :   "mem.zoneinfo.nr_zone_inactive_anon"
+        })
+    ZONEINFO_PER_NODE.update({
+        "nr_shmem_hugepages"            :   "mem.zoneinfo.nr_shmem_hugepages",
+        "nr_shmem_pmdmapped"            :   "mem.zoneinfo.nr_shmem_pmdmapped",
+        "nr_file_hugepages"             :   "mem.zoneinfo.nr_file_hugepages",
+        "nr_file_pmdmapped"             :   "mem.zoneinfo.nr_file_pmdmapped",
+        "nr_kernel_misc_reclaimable"    :   "mem.zoneinfo.nr_kernel_misc_reclaimable"
+    })
+
 
 if __name__ == '__main__':
     try:
         opts = ZoneinfoOptions()
         mngr = pmcc.MetricGroupManager.builder(opts,sys.argv)
         opts.context = mngr.type
+
+        # After this VERSION new zoneifo metric has been introduced so checking the current version,
+        # if it is the latest version of pcp then get all the data otherwise drop the metrics
+        # which will not be there in older version.
+        # this change make pcp-zoneinfo comptabile with rhel7 and ol7 where VERSION 4.x.x being used.
+        VERSION="5.0.0"
+        current_version=pmapi.pmContext.pmGetConfig("PCP_VERSION")
```

Comment:
@orasagar this is looking at the local installed PCP version though.  What if we're replaying an archive or connected to a remote host with different version?  A better approach is to attempt to use the new metric, and fallback to the other form if the new metric doesn't exist.

Reply:
@natoscott that sound reasonable ,will fix it and get back